tests/conftest.py

```diff
@@ -225,6 +196,67 @@ def set_basic_auth_logins(self, **jira_class_kwargs):
             self.jira_sysadmin = JIRA(self.CI_JIRA_URL, **jira_class_kwargs)
             self.jira_normal = JIRA(self.CI_JIRA_URL, **jira_class_kwargs)
 
+    def _project_exists(self, project_key: str) -> bool:
+        try:
+            self.jira_admin.project(project_key)
+        except Exception as e:  # If the project does not exist a warning is thrown
+            if "No project could be found" in str(e):
+                return False
+        return True
+
+    def _remove_project(self, project_key):
+        """Ensure if the project exists we delete it first"""
+
+        wait_between_checks_secs = 2
+        time_to_wait_for_delete_secs = 40
+        wait_attempts = int(time_to_wait_for_delete_secs / wait_between_checks_secs)
+
+        # TODO(ssbarnea): find a way to prevent SecurityTokenMissing for On Demand
+        # https://jira.atlassian.com/browse/JRA-39153
+        try:
+            self.jira_admin.project(project_key)
+        except Exception as e:  # If the project does not exist a warning is thrown
+            if "No project could be found" not in str(e):
+                raise e
+        else:
+            # if no error is thrown that means the project exists, so we try to delete it
+            try:
+                self.jira_admin.delete_project(project_key)
+            except Exception as e:
+                LOGGER.warning("Failed to delete %s\n%s", project_key, e)
+
+        # wait for the project to be deleted
+        for _ in range(1, wait_attempts):
+            if not self._project_exists(project_key):
+                # If the project does not exist a warning is thrown
+                # so once this is raised we know it is deleted successfully
+                break
+            sleep(wait_between_checks_secs)
+
+        if self._project_exists(project_key):
+            raise TimeoutError(
+                " Project '{project_key}' not deleted after {time_to_wait_for_delete_secs} seconds"
+            )
+
+    def _create_project(
+        self, project_key: str, project_name: str, allow_exist: bool = False
```

Comment:
I'd name this `allow_exist` differently .. had to look up the code as to what you exactly meant by it.
- force_recreate
- remove_if_exists

a few suggestions => everything is subjective of course

Reply:
force_recreate sounded good to me